NYT-SwiftUI/Network/NetworkLayer.swift

```diff
@@ -41,4 +46,73 @@ class NetworkLayer {
             }
         }
     }
+    
+    static func getSports(completionHandler completion: @escaping ([News]?, Bool) -> Void) {
+        getData(Constants.NewsTypesForURL.sports) {
+            newsList, isAlldataLoaded in
+            DispatchQueue.main.async {
+                completion(newsList, isAlldataLoaded)
+            }
+        }
+    }
+    
+    static func getScience(completionHandler completion: @escaping ([News]?, Bool) -> Void) {
+        getData(Constants.NewsTypesForURL.science) {
+            newsList, isAlldataLoaded in
+            DispatchQueue.main.async {
+                completion(newsList, isAlldataLoaded)
+            }
+        }
+    }
+}
+
+//MARK: - Firebase Management
+
+extension NetworkLayer {
+    
+    static func getAllDataFromDatabase(completionHandler completion: @escaping ([News]?, Bool) -> Void) {
+        firebaseRef.getData {
+            error, snapshot in
+            var newsList: [News] = []
+            if let data = snapshot?.value as? [String: Any] {
+                for (_, item) in data {
+                    let article = try? FirebaseDataDecoder().decode(News.self, from: item)
+                    if let article = article {
+                        newsList.append(article)
+                    }
+                }
+                completion(newsList, true)
+            }
+        }
+    }
+    
+    static func insertIntoDatabase(_ newsArticle: News) {
+        let data = try? JSONEncoder().encode(newsArticle)
+        if let data = data {
+            let json = try? JSONSerialization.jsonObject(with: data)
+            firebaseRef.childByAutoId().setValue(json)
+        }
+    }
+    
+    static func removeFromDatabase(_ newsArticle: News) {
+        firebaseRef.getData {
+            error, snapshot in
+            var key: String?
+            let newsList = snapshot?.value as? [String: Any]
+            if let newsList = newsList {
+                for item in newsList {
```

Comment:
use contains function instead of traversing news list and use it throughout the app.